server/bleep/src/webserver.rs

```diff
@@ -55,6 +56,21 @@ pub async fn start(app: Application) -> anyhow::Result<()> {
         .route("/index", get(index::handle))
         // repo management
         .nest("/repos", repos::router())
+        // docs management
+        .nest(
+            "/docs",
+            Router::new()
+                .route("/", get(docs::list)) // list all doc providers
+                .route("/search", get(docs::search)) // text search over doc providers
+                .route("/sync", get(docs::sync)) // index a new doc provider
+                .route("/verify", get(docs::verify)) // verify if a doc url is valid
+                .route("/:id", get(docs::list_one)) // list a doc provider by id
+                .route("/:id", delete(docs::delete)) // delete a doc provider by id
+                .route("/:id/resync", get(docs::resync)) // resync a doc provider by id
+                .route("/:id/search", get(docs::search_with_id)) // search/list sections of a doc provider
+                .route("/:id/list", get(docs::list_with_id)) // list pages of a doc provider
```

Comment:
```suggestion
                .route("/:id/pages", get(docs::list_with_id)) // list pages of a doc provider
```

Reply:
have omitted api nomenclature changes for now, as they require a FE & BE change.